### ISSUE TITLE
failed to build joystream from source

### ISSUE THREAD

**Issue by clong12** [Joystream/substrate-node-joystream#86]
Hi,
I followed the guide https://github.com/Joystream/substrate-node-joystream to build the joystream node from source on centos( CentOS Linux release 7.7.1908 (Core)), and I got the error:
warning: static item is never used: `MSG_NO_ACTOR_FOR_ROLE`
  --> substrate-runtime-joystream/src/roles/actors.rs:14:1
   |
14 | static MSG_NO_ACTOR_FOR_ROLE: &str = "For the specified role, no actor is currently staked.";
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

error[E0063]: missing field `forum` in initializer of `joystream_node_runtime::GenesisConfig`
   --> src/chain_spec.rs:176:5
    |
176 |     GenesisConfig {
    |     ^^^^^^^^^^^^^ missing `forum`

error[E0063]: missing field `forum` in initializer of `joystream_node_runtime::GenesisConfig`
   --> src/chain_spec.rs:278:5
    |
278 |     GenesisConfig {
    |     ^^^^^^^^^^^^^ missing `forum`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0063`.
error: could not compile `joystream-node`.

To learn more, run the command again with --verbose.

Does anyone can help? Thank you very much.

**Comment by mnaamani** [comment]
I've added some extra steps you can take to build successfully here https://github.com/Joystream/substrate-node-joystream/issues/81#issuecomment-552795815

**Comment by clong12** [comment]
@mnaamani 
Thank you for reply. I have built joystream_node successfully. When I run the node "joystream_node' without any option, it work well. However, when I run the node passing the parameter --key, it shows the error:
joystream-node --validator --key mykey_address --name mynode_name
2019-11-13 17:34:46 Joystream Node
2019-11-13 17:34:46   version 1.0.0-x86_64-linux-gnu
2019-11-13 17:34:46   by Joystream, 2019
2019-11-13 17:34:46 Chain specification: Joystream Testnet v2
2019-11-13 17:34:46 Node name: mynode_name
2019-11-13 17:34:46 Roles: AUTHORITY
Error: Error(Keystore(InvalidSeed), State { next_error: None, backtrace: InternalBacktrace { backtrace: None } })
I have created the key via https://testnet.joystream.org/acropolis/pioneer/#/accounts.
